### DISCUSSION TITLE
Can Inertia Link "reload" @routes directive?

### DISCUSSION THREAD

**Discussion by maxonfjvipon** [inertiajs/inertia#915] [Help (React)] [answered]
Hey everyone.
My stack is Laravel and React. I use Inertia and [mcamara/laravel-localization](https://github.com/mcamara/laravel-localization).
This is my web.php (see [here](https://github.com/mcamara/laravel-localization#usage) how it works)
```
Route::prefix(LaravelLocalization::setLocale())
    ->middleware('localizationRedirect')
    ->group(function () {
        // routes
    });
```
Also on my front-end at layout I have locale switcher:
```
export const LocaleDropdown = () => {
    const {locales} = usePage().props

    return (
        <Dropdown key="lang-change" arrow trigger={['click']} overlay={
            <Menu>
                {locales.supported.map(locale => (
                    <Menu.Item key={locale}>
                        <Link
                            preserveState
                            preserveScroll
                            method="GET"
                            href={locales.current_localized[locale]}
                            only={['locales']}
                        >
                            {locale.toUpperCase()}
                        </Link>
                    </Menu.Item>
                ))}
            </Menu>
        }>
            <Button>
                Change 
            </Button>
        </Dropdown>
    )
}
```
Also on front-end I use ```route``` helper function for requests.
When I choose a specific locale, I visit to https://host/{locale}/current-path. Actually it's just a reloading current page with new locale. ```locales``` is a shared variable, that contains supported locales and current url with those supported locales.
When I reload the page via Inertia link with new locale I can see, that interface is translated, BUT if I press some button or link, that uses ```route``` function to get route path, I can see, that this path does not contain current locale. For example
1. Current locale - en
2. Interface lang is English
3. ```console.log(route('register')) // http://host/register```
4. Switch locale to es <- visit to http://host/es/current-path <- click ```<Link href="http://host/es/current-path">es</Link>```
5. Now app()->getLocale() - 'es'
6. Interface lang is Spain
7. But ```console.log(route('register'))``` is still http://host/register. Should be http://host/es/register.

Only If I press F5 (full page reloading), then ```route('register')``` is http://host/es/register.

If instead on Inertia links I use simple ```<a>``` links to switch locales, then routes are correct and contain current locale.
Also I tried to remove ```preserveState``` and ```only``` props - the same.
So it seems to me that Inertia links does not reload ```@routes``` directive in app.blade.php. Maybe I'm wrong and there is no connection with directive, but still, I would want to use ```route``` function instead of my custom one.

If anybody needs, my custom ```tRoute``` is below:
```
export const useTransRoutes = () => {
    const {locales} = usePage().props

    const tRoute = (name, ...params) => {
        const splitUrl = route(name, ...params).split("/")
        const locale = splitUrl[3] // locale from url

        splitUrl.splice(3, locales.supported.includes(locale) ? 1 : 0)
        if (locales.default !== locales.current) {
            splitUrl.splice(3, 0, locales.current)
        }

        return splitUrl.join('/')
    }

    return {
        tRoute
    }
}
```

**Comment by ajnsn** [comment, marked as answer]
@maxonfjvipon Inertia is more an SPA which means there are typically no full page reloads and not all parts of page will always get re-rendered. I assume that your `route()` helper is only rendered on page load. You will need a more dynamic approach, e.g. passing the routes via shared props, in the respective Inertia responses or fetching the routes from an endpoint after language change.

**Reply by maxonfjvipon** [reply]
Thanks for your answer. 
I think I'll figure this out
Also leave here this [one](https://github.com/tighten/ziggy/issues/436)
